cupyx/jit/_preprocess.py

```diff
@@ -0,0 +1,47 @@
+def _strip_lambda(source):
```

Comment:
* How about walking an AST as suggested in my original comment, instead of relying on heuristics?

```py
def get_lambda_node(x):
    lines = linecache.getlines(inspect.getsourcefile(x))
    _, lineno = inspect.getsourcelines(x)
    lambdas = [n for n in ast.walk(ast.parse(''.join(lines))) if isinstance(n, ast.Lambda) and n.lineno == lineno]
    if len(lambdas) != 1: raise RuntimeError
    return lambdas[0]
```

* Maybe better to memorize the result?